pulsar-client-api/src/main/java/org/apache/pulsar/client/api/KeySharedPolicy.java

```diff
@@ -29,7 +30,8 @@
  */
 @InterfaceAudience.Public
 @InterfaceStability.Stable
-public abstract class KeySharedPolicy {
+public abstract class KeySharedPolicy implements Serializable {
+    private static final long serialVersionUID = 1L;
```

Comment:
Add `private static final long serialVersionUID = 1L;`  to the implementation classes of the abstract `KeySharedPolicy` class instead of adding it to the abstract class where it is unnecessary.

Reply:
Yes, my bad! Made these changes.